src/logic/tokens/store/actions/fetchSafeTokens.ts

```diff
@@ -11,12 +11,21 @@ import { makeToken } from 'src/logic/tokens/store/model/token'
 import { TOKEN_REDUCER_ID } from 'src/logic/tokens/store/reducer/tokens'
 import updateSafe from 'src/routes/safe/store/actions/updateSafe'
 import { SAFE_REDUCER_ID } from 'src/routes/safe/store/reducer/safe'
+import { Dispatch } from 'redux'
+import { backOff } from 'exponential-backoff'
 
 const humanReadableBalance = (balance, decimals) => new BigNumber(balance).times(`1e-${decimals}`).toFixed()
 const noFunc = () => {}
 const updateSafeValue = (address) => (valueToUpdate) => updateSafe({ address, ...valueToUpdate })
 
-const fetchSafeTokens = (safeAddress) => async (dispatch, getState) => {
+let isFetchingData = false
+
+const fetchSafeTokens = (safeAddress: string) => async <T extends () => unknown>(
```

Comment:
So this action returns a function with a type argument which is not used anywhere in the app. Also type of getState should be `() => State` I think

Reply:
Also type definitions are inconsistent (e.g this is the only action which uses this kind of typing). I think redux is not very typescript friendly by itself, so maybe we need to look into something like https://github.com/piotrwitek/typesafe-actions in the future

Reply:
If I put `() => State` I would need to type the entire state or I am wrong?

Reply:
We already have a type for the state, can’t you just import it from somewhere? IIRC redux even has a type for `GetState` which accepts are type argument so you can use it like this `GetState<State>`

Reply:
I couldn't find the type that you mention, just created a new one that needs to be further improved with all the state types.